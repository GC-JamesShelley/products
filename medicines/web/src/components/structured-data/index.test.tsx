import { shallow } from 'enzyme';
import React from 'react';
import {
  DrugListStructuredData,
  DrugStructuredData,
  SubstanceListStructuredData,
  SubstanceStructuredData,
} from './index';

describe(SubstanceStructuredData, () => {
  it('should render', () => {
    const component = shallow(
      <SubstanceStructuredData substanceName={'My cool substance'} />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe(SubstanceListStructuredData, () => {
  it('should render', () => {
    const component = shallow(
      <SubstanceListStructuredData
        substanceNames={['First cool substance', 'Second cool substance']}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe(DrugStructuredData, () => {
  it('should render', () => {
    const component = shallow(<DrugStructuredData drugName={'My cool drug'} />);
    expect(component).toMatchSnapshot();
  });
});

describe(DrugListStructuredData, () => {
  it('should render', () => {
    const component = shallow(
      <DrugListStructuredData
        drugNames={['First cool drug', 'Second cool drug']}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
