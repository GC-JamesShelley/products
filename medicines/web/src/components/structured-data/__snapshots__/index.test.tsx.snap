// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugListStructuredData should render 1`] = `
<JsonLd
  item={
    Object {
      "@context": "https://schema.org",
      "@type": "ItemList",
      "itemListElement": Array [
        Object {
          "@type": "ListItem",
          "item": Object {
            "@type": "Drug",
            "name": "First cool drug",
            "url": "https://products.mhra.gov.uk/product?product=First%20cool%20drug",
          },
          "position": 0,
        },
        Object {
          "@type": "ListItem",
          "item": Object {
            "@type": "Drug",
            "name": "Second cool drug",
            "url": "https://products.mhra.gov.uk/product?product=Second%20cool%20drug",
          },
          "position": 1,
        },
      ],
    }
  }
/>
`;

exports[`DrugStructuredData should render 1`] = `
<JsonLd
  item={
    Object {
      "@context": "https://schema.org",
      "@type": "Drug",
      "name": "My cool drug",
    }
  }
/>
`;

exports[`SubstanceListStructuredData should render 1`] = `
<JsonLd
  item={
    Object {
      "@context": "https://schema.org",
      "@type": "ItemList",
      "itemListElement": Array [
        Object {
          "@type": "ListItem",
          "item": Object {
            "@type": "Substance",
            "name": "First cool substance",
            "url": "https://products.mhra.gov.uk/substance?substance=First%20cool%20substance",
          },
          "position": 0,
        },
        Object {
          "@type": "ListItem",
          "item": Object {
            "@type": "Substance",
            "name": "Second cool substance",
            "url": "https://products.mhra.gov.uk/substance?substance=Second%20cool%20substance",
          },
          "position": 1,
        },
      ],
    }
  }
/>
`;

exports[`SubstanceStructuredData should render 1`] = `
<JsonLd
  item={
    Object {
      "@context": "https://schema.org",
      "@type": "Substance",
      "name": "My cool substance",
    }
  }
/>
`;
